azkaban-common/src/main/java/azkaban/executor/ExecutorManager.java

```diff
@@ -1776,12 +1776,49 @@ private void processFlow(ExecutionReference reference,
       }
     }
 
+    /* Helper method to fetch  overriding Executor, if a valid user has specifed otherwise return null */
```

Comment:
We return null as we want to default to normal behaviour of asking selector to pick an executor. Though I have added more log to identify different exception scenarios.